code/parse/sexp_container.cpp

```diff
@@ -787,3 +787,261 @@ const char *sexp_container_CTEXT(int node)
 	// we've hit sanity limit; give up
 	return Empty_str;
 }
+
+// status SEXPs
+int sexp_is_container_empty(int node)
+{
+	const char *container_name = CTEXT(node);
+	const auto *p_container = get_sexp_container(container_name);
+
+	if (!p_container) {
+		Warning(LOCATION, "Is-container-empty called on nonexistent container %s.", container_name);
+		return SEXP_FALSE;
+	}
+
+	const auto &container = *p_container;
+
+	if (container.is_map()) {
+		if (container.map_data.empty()) {
+			return SEXP_TRUE;
+		}
+	} else if (container.is_list()) {
+		if (container.list_data.empty()) {
+			return SEXP_TRUE;
+		}
+	} else {
+		Error(LOCATION,
+			"Unknown container type. Container %s is of type %d and this is not a valid type",
+			container_name,
+			(int)container.type);
+	}
+
+	return SEXP_FALSE;
+}
+
+int sexp_get_container_size(int node)
+{
+	const char *container_name = CTEXT(node);
+	const auto *p_container = get_sexp_container(container_name);
+
+	if (!p_container) {
+		Warning(LOCATION, "Get-container-size called on nonexistent container %s.", container_name);
+		return 0;
```

Comment:
In contrast, this can definitely stay as-is, since 0 makes sense in any situation.  The mission designer is going to have to fix it any way.

So far the other instances can stay as-is, too.